### PR TITLE
 Add Go linter to CI.

### PR DESCRIPTION
**Pull request by LuisPH3**
This PR enables go linting in PR CI.

Both [vet](https://pkg.go.dev/cmd/vet) and [staticcheck](https://staticcheck.dev/docs/) are to be satisficed before PRs are mergeable.

### CODE REVIEW

File: BUILD.md [hunk 1]
```diff
@@ -270,3 +270,21 @@ The ct Evm.StepN interface is used to evaluate N instructions in different EVM i
...
+
+
+## Static analysis
+
+Tosca project uses static analysis to improve code quality.
+
+### Go
+
+Currently, two different static analysis tools are used:
+- [Vet](https://pkg.go.dev/cmd/vet) Is the default go linter, bundled with the toolchain. If finds rather critical bugs that should be addressed right away. To execute it run `go vet ./go/...`. This tool is currently enabled in the CI, and pull request will be prevented from merging when checks are not satisfied.
```
Comment (HerbertJordan) [inline BUILD.md:282]:
```suggestion
- [Vet](https://pkg.go.dev/cmd/vet) Is the default go linter, bundled with the toolchain. It finds rather critical bugs that should be addressed right away. To execute it run `go vet ./go/...`. This tool is currently enabled in the CI, and pull request will be prevented from merging when checks are not satisfied.
```

File: BUILD.md [hunk 2]
```diff
@@ -270,3 +270,21 @@ The ct Evm.StepN interface is used to evaluate N instructions in different EVM i
...
+- [StaticCheck](https://staticcheck.dev/) Finds other [errors](https://staticcheck.dev/docs/checks/) related to styling and good practices. 
+
+Lints can be disabled as described [here](https://golangci-lint.run/usage/false-positives/). Please use `//nolint:pass` where pass is the linter pass that discovers the error. Do not ignore all passes as one line may hide more than one error.
+
+### C++
+
+There is no current static analysis for the evmzero. Integrating clang-tidy should not be too difficult.
```
Comment (HerbertJordan) [inline BUILD.md:289]:
```suggestion
There is current no static analysis for the the C++ code in this project.
```
It is not very useful for any reader to state that the integration of `clang-tidy` should not be too hard. I, as a reader, would not go and implement it, nor would I believe that it is "not too difficult", otherwise it would have been implemented.

**Review by HerbertJordan — APPROVED** [review]

**Review by facuMH — DISMISSED** [review]

**Review by facuMH — DISMISSED** [review]

File: cpp/vm/evmzero/evmzero.cc [hunk 3]
```diff
@@ -106,7 +110,18 @@ class VM : public evmc_vm {
             .name = "evmzero",
             .version = "0.1.0",
 
-            .destroy = [](evmc_vm* vm) { delete static_cast<VM*>(vm); },
+            .destroy =
+                [](evmc_vm* vm) {
+                  delete static_cast<VM*>(vm);
+// This seems a little hacky, but this is property that requires special compilation:
+// When running coverage, records are keep in memory and dumped at the end of the execution,
```
Comment (simonlechner) [inline cpp/vm/evmzero/evmzero.cc:117]:
```suggestion
// When running coverage, records are kept in memory and dumped at the end of the execution,
```

**Review by simonlechner — DISMISSED** [review]

File: cpp/vm/evmzero/evmzero.cc [hunk 4]
```diff
@@ -23,6 +22,11 @@
 #include "vm/evmzero/profiler.h"
 #include "vm/evmzero/sha3_cache.h"
 
+#if defined(TOSCA_COVERAGE)
+// Since Gcc 11, before __gcov_flush
+extern "C" void __gcov_dump();
```
Comment (HerbertJordan) [inline cpp/vm/evmzero/evmzero.cc:27]: There might be a merge issue here. This code is part of the code coverage PR for C++ code and should not be included in the PR adding Go linter support.

**Review by HerbertJordan — CHANGES_REQUESTED** [review]

File: cpp/vm/evmzero/evmzero.cc (hunk 4 above)
Comment (LuisPH3) [reply cpp/vm/evmzero/evmzero.cc]: sorry about that, pretty bad rebase.

File: cpp/vm/evmzero/evmzero.cc (hunk 3 above)
Comment (LuisPH3) [reply cpp/vm/evmzero/evmzero.cc]: code was removed

**Review by facuMH — APPROVED** [review]

**Review by HerbertJordan — APPROVED** [review]

**Review by simonlechner — APPROVED** [review]
